hr_public_holidays/models/hr_public_holidays.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding:utf-8 -*-
+#
+#
+#    Copyright (C) 2011,2013 Michael Telahun Makonnen <mmakonnen@gmail.com>.
+#    All Rights Reserved.
+#    Copyright (C) 2014 initOS GmbH & Co. KG (<http://www.initos.com>).
+#    Author Nikolina Todorova <nikolina.todorova@initos.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published
+#    by the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#
+
+from datetime import date
+from openerp import fields, models, api, _
+from openerp.exceptions import ValidationError
+
+
+class HrHolidays(models.Model):
+    _name = 'hr.holidays.public'
+    _description = 'Public Holidays'
+    _rec_name = 'year'
+    _order = "year"
+
+    year = fields.Integer(
+        required=True,
+        default=date.today().year
+    )
+    line_ids = fields.One2many(
+        'hr.holidays.public.line',
+        'year_id',
+        'Holiday Dates'
+    )
+    country_id = fields.Many2one(
+        'res.country',
+        'Country'
+    )
+
+    _sql_constraints = [
+        ('year_unique',
+         'UNIQUE(year,country_id)',
+         _('Duplicate year and country!')),
+    ]
+
+    @api.one
+    @api.constrains('year')
+    def _check_year(self):
+        if self.search_count([('year', '=', self.year),
+                              ('country_id', '=', False),
+                              ('id', '!=', self.id)]):
+            raise ValidationError('You can\'t create duplicate year')
+        return True
```

Comment:
It is need as the constraint is not triggered in cases that no country is supplied... will allow duplicate years as long as you do not supply country